src/DbPostgreSQLAdapter/Database.cpp

```diff
@@ -0,0 +1,134 @@
+#include "stdafx.h"
+
+#include "Database.h"
+#include "DbAdapterInterface/ITable.h"
+#include "RecordSet.h"
+#include "Table.h"
+#include "TableRecordSet.h"
+#include "Transaction.h"
+
+#include <libpq-fe.h>
+
+namespace {
+	std::unique_ptr<PGresult, void(*)(PGresult*)> createRAIIPGresult(PGresult* result)
+	{
+		return std::unique_ptr<PGresult, void(*)(PGresult*)>(result, PQclear);
+	}
+}
+
+namespace systelab::db::postgresql {
+
+	Database::Database(PGconn* database)
+		: m_database(database)
+		, m_lastOperationRowsAffected(0)
+		, m_lastInsertedRowId(0)
+	{
+	}
+
+	Database::~Database()
+	{
+		PQfinish(m_database);
+	}
+
+	ITable& Database::getTable(std::string tableName)
+	{
+		const auto itr = m_tables.find(tableName);
+		if (itr == m_tables.cend())
```

Comment:
Sounds odd to me that you would create the Table as part of the getTable method, but if that's the approach then don't overcomplicate it:
```
auto itr = m_tables.find(tableName);
if (itr == m_tables.end())
{
   auto& insertedValue = m_tables[tableName] = std::make_unique<Table>(*this, tableName)
    return *insertedValue;
}
return **itr;
```

Reply:
I'll take care of it